sdk/core/core-lro/src/poller/poller.ts

```diff
@@ -143,39 +159,42 @@ export function buildCreatePoller<TResponse, TResult, TState extends OperationSt
             inputAbortSignal?.removeEventListener("abort", abortListener);
           }
           if (resolveOnUnsuccessful) {
-            return poller.getResult() as TResult;
+            return poller.result as TResult;
           } else {
             switch (state.status) {
               case "succeeded":
-                return poller.getResult() as TResult;
+                return poller.result as TResult;
               case "canceled":
                 throw new Error(cancelErrMsg);
               case "failed":
-                throw state.error;
+                throw state!.error;
               case "notStarted":
               case "running":
                 throw new Error(`Polling completed without succeeding or failing`);
             }
           }
         })().finally(() => {
           resultPromise = undefined;
-        })),
-      async poll(pollOptions?: { abortSignal?: AbortSignalLike }): Promise<void> {
+        });
+        return resultPromise;
+      },
+      async poll(pollOptions?: { abortSignal?: AbortSignalLike }): Promise<TState> {
+        await statePromise;
         if (resolveOnUnsuccessful) {
-          if (poller.isDone()) return;
+          if (poller.isDone) return state!;
         } else {
           switch (state.status) {
             case "succeeded":
-              return;
+              return state!;
             case "canceled":
               throw new Error(cancelErrMsg);
             case "failed":
-              throw state.error;
+              throw state!.error;
           }
         }
         await pollOperation({
           poll,
-          state,
+          state: state!,
```

Comment:
I'd prefer if we nullchecked state before calling `pollOperation` and throw an actionable error